src/utils/BlockUtils.ts

```diff
@@ -166,3 +169,53 @@ export class BlockFinder {
     }
   }
 }
+
+/**
+ * @notice Get the block number for a given timestamp fresh from on-chain data if not found in redis cache.
+ * If redis cache is not available, then requests block from blockFinder.
+ * @param chainId Chain to load block finder for.
+ * @param timestamp Approximate timestamp of the to requested block number.
+ * @param blockFinder Caller can optionally pass in a block finder object to use instead of creating a new one
+ * or loading from cache. This is useful for testing primarily.
+ * @returns Block number for the requested timestamp.
+ */
+export async function getCachedBlockForTimestamp(
+  chainId: number,
+  timestamp: number,
+  blockFinder: BlockFinder,
+  cache?: CachingMechanismInterface
+): Promise<number> {
+  // Resolve a convenience function to directly compute what we're
+  // looking for.
+  const resolver = async () => (await blockFinder.getBlockForTimestamp(timestamp)).number;
+
+  // If no redis client, then request block from blockFinder.
+  if (!isDefined(cache)) {
+    return resolver();
+  }
+  // Cache exists. We should first check if it's possible to retrieve the block number from cache.
+  else {
```

Comment:
This else clause seems redundant - OK to drop it to reduce the nesting here?

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/438/commits/147a409aef4f77eb6d5a5695550ccd93206d39fe